### ISSUE TITLE
gemma models have low PCC

### ISSUE THREAD

**Issue by kmabeeTT** [tenstorrent/tt-xla#2344]
Results are gathered locally using nov23 commit - later tt-xla is a bit unstable in general, lots of models recently broke.

Try cherry-picking the workaround from #2333 ("Perform accuracy comparisons in Float64") via `git cherry-pick -x 76134222a65d146e245ea80d49949ae1a1fb26fc`

Most important one to get passing is this, but it requires blackhole p150 since it's a 7B param model:

```
python ~/scripts/compare_pcc.py  gemma_tests_pcc_float32_commit_2f603024.xml gemma_tests_pcc_float64_commit_2f603024.xml
test_name                                                                                        pcc_before  pcc_after  pcc_change
test_all_models_torch[gemma/codegemma/pytorch-google/codegemma-2b-single_device-full-inference]    1.007873   0.989519   -0.018354
test_all_models_torch[gemma/pytorch-google/gemma-1.1-2b-it-single_device-full-inference]           1.020076   0.987846   -0.032230
test_all_models_torch[gemma/pytorch-google/gemma-1.1-7b-it-single_device-full-inference]           0.941214   0.925659   -0.015554
test_all_models_torch[gemma/pytorch-google/gemma-2-2b-it-single_device-full-inference]             1.030244   0.997748   -0.032496
test_all_models_torch[gemma/pytorch-google/gemma-2-9b-it-single_device-full-inference]             1.027682   0.999463   -0.028220
test_all_models_torch[gemma/pytorch-google/gemma-2b-single_device-full-inference]                  0.907602   0.883016   -0.024587

Some others (can run on n150 wormhole):

**Comment by kmabeeTT** [comment]
FWIW if I trim number of layers in 7B model from 28 to 14 or 7 to see it "fit" on n150, PCC is 0.99 (and it jumps to 0.99 on BH too).

```
           new_num_layers = 7 
           model.model.layers = torch.nn.ModuleList(
                list(model.model.layers)[:new_num_layers]
            )
            model.config.num_hidden_layers = new_num_layers
            return model

```

**Comment by sonalibaskaran2499** [comment]
@kmabeeTT 
Re-tested Gemma-1.1-7B-it with padding = true and apply_chat_template:
pcc=1.0563201904296875

**Logs**

[gemma_1.1.7b-it.log](https://github.com/user-attachments/files/24190920/gemma_1.1.7b-it.log)

PR - https://github.com/tenstorrent/tt-forge-models/pull/348
